scripts/dts/extract/compatible.py

```diff
@@ -43,6 +43,18 @@ def extract(self, node_address, prop, def_label):
                 compat_defs: "1",
             }
             insert_defs(node_address, load_defs, {})
+
+            # Generate defines of the form:
+            # #define DT_<COMPAT>_SOC_<SOC LABEL> 1
+            if 'soc-label' in reduced[node_address]['props'].keys():
```

Comment:
`x in dict` in Python always tests the keys, so could drop the `.keys()`.

Could do this too:

```python
soc_label = reduced[node_address]['props'].get('soc-label')
if soc_label is not None:
    soc_instance = ...
```